config/config.go

```diff
@@ -104,6 +104,71 @@ func NewViper() *viper.Viper {
 	return viper.NewWithOptions(viper.KeyDelimiter(ViperDelimiter))
 }
 
+// fromStringMap creates a Viper object from a map[string]interface{}
+func fromStringMap(cfg map[string]interface{}) *viper.Viper {
+	viperCfg := NewViper()
+	viperCfg.MergeConfigMap(cfg)
+	return viperCfg
+}
+
+// deepSearch scans deep maps, following the key indexes listed in the
+// sequence "path".
+// The last value is expected to be another map, and is returned.
+//
+// In case intermediate keys do not exist, or map to a non-map value,
+// a new map is created and inserted, and the search continues from there:
+// the initial map "m" may be modified!
+// This function comes from Viper code https://github.com/spf13/viper/blob/5253694/util.go#L201-L230
+// It is used here because of https://github.com/spf13/viper/issues/819
+func deepSearch(m map[string]interface{}, path []string) map[string]interface{} {
+	for _, k := range path {
+		m2, ok := m[k]
+		if !ok {
+			// intermediate key does not exist
+			// => create it and continue from there
+			m3 := make(map[string]interface{})
+			m[k] = m3
+			m = m3
+			continue
+		}
+		m3, ok := m2.(map[string]interface{})
+		if !ok {
+			// intermediate key is a value
+			// => replace with a new map
+			m3 = make(map[string]interface{})
+			m[k] = m3
+		}
+		// continue search from here
+		m = m3
+	}
+	return m
+}
+
+// toStringMap creates a map[string]interface{} from a Viper object
+// It is equivalent to v.AllSettings() but it maps nil values
+// It is used here because of https://github.com/spf13/viper/issues/819
+func toStringMap(v *viper.Viper) map[string]interface{} {
```

Comment:
This can be simplified if/once spf13/viper#819 gets resolved (it seems like Bogdan already stumbled on this one 😄). The code is the same as the `AllSettings` method except that it does not skip over `nil` values.

Reply:
Not used anymore, `cast.ToStringMap` seems to overcome this issue